libpod/healthcheck.go

```diff
@@ -73,20 +61,30 @@ func (c *Container) runHealthCheck() (define.HealthCheckStatus, error) {
 	if len(newCommand) < 1 || newCommand[0] == "" {
 		return define.HealthCheckNotDefined, errors.Errorf("container %s has no defined healthcheck", c.ID())
 	}
-	captureBuffer := bufio.NewWriter(&capture)
-	hcw := hcWriteCloser{
-		captureBuffer,
+	rPipe, wPipe, err := os.Pipe()
+	if err != nil {
+		return define.HealthCheckInternalError, errors.Wrapf(err, "unable to create pipe for healthcheck session")
 	}
+	defer wPipe.Close()
+	defer rPipe.Close()
+
 	streams := new(define.AttachStreams)
-	streams.OutputStream = hcw
-	streams.ErrorStream = hcw
 
 	streams.InputStream = bufio.NewReader(os.Stdin)
-
+	streams.OutputStream = wPipe
```

Comment:
Is this safe? Seems there's a risk that the two can write at the same time and result in corrupted, mixed output

Reply:
It is no different than looking at the output in your terminal, no?

Reply:
I wanted different pipes but i tried to keep behavior same as the previous code. But added different pipe for `stderror` cause that looks better code wise.

Reply:
This will now hide either stdout or stderr, I am not sure if it is the right thing to do different pipes. What does docker do?

Reply:
@Luap99 I am not sure. I'll check but i guess sharing same pipe seems to be safest behavior wise. The bigger problem is i am not getting any logs on CI while i get it when i run container on my local, is CI limited ?

Reply:
Lets now i have copied retry logic from some of the existing tests.

Reply:
Marking this as resolved. Since we are interested in terminal output and sharing pipe/buffer is the same behavior as we had in previous implementation which should not cause any risk.

Please feel free to unresolved this if needed.